app/src/main/kotlin/app/vitune/android/database/mapper/PipedSessionMapper.kt

```diff
@@ -0,0 +1,26 @@
+package app.vitune.android.database.mapper

```

Comment:
I don't understand how this makes the current DB system any better. Does it allow for a later module extraction? If so, do you expect someone to come up with a desktop app written in KT?

Reply:
The current app is written solely for android. So the Database it uses uses an android specific DB that may not be compatible with other database technologies. As the entities and its annotations are also specific to that android db you need a layer of indirection. This is what this is doing. We split the application code from the db specific code and make the application code independent from the platform specific code, enabling us to develop an application with multiple clients or uis if you will so.

Reply:
I like the enthusiasm here. Thank you for your idea. But that's not what I meant: do you imagine implementing such an app in kotlin or a different programming language?

Reply:
ill probably just use the kotlin composite framework. I dont really fancy building an interface to another programming language.